src/script/script.h

```diff
@@ -411,6 +411,7 @@ class CScript : public CScriptBase
     CScript(const_iterator pbegin, const_iterator pend) : CScriptBase(pbegin, pend) { }
     CScript(std::vector<unsigned char>::const_iterator pbegin, std::vector<unsigned char>::const_iterator pend) : CScriptBase(pbegin, pend) { }
     CScript(const unsigned char* pbegin, const unsigned char* pend) : CScriptBase(pbegin, pend) { }
+    CScript(const CScript& a) : CScript(a.begin(), a.end()) {}
```

Comment:
Might want to add a comment describing why this explicit copy constructor (which seems to be redundant) was added, otherwise it might be removed again with the next 'clean up warnings' PR.

Reply:
Agreed! That is also why it is important to remove the UBSan suppressions as part of this PR: to catch the consequence non-refactoring nature of such a mistaken change :)

These needs to go:

https://github.com/bitcoin/bitcoin/blob/b8f041af2d2c89afdc14f8c632dfb7fecb535cb5/test/sanitizer_suppressions/ubsan#L1-L2

Reply:
Added a comment and removed the suppressions.